gradle/generate-pom.gradle

```diff
@@ -261,33 +317,181 @@ class ProjectDependenciesAsXml {
     private static boolean isExternal(Dependency dependency) {
         return AbstractExternalModuleDependency.isAssignableFrom(dependency.class)
     }
+}
+
+/**
+ * A project dependency with its scope.
+ *
+ * @see
+ * <a href="https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Scope">
+ *     More on dependency scopes </a>.
+ */
+class DependencyWithScope implements Comparable<DependencyWithScope> {
 
-    private static void addIfNotPresent(Dependency dependency, Set<Dependency> dependencySet) {
-        boolean isDuplicate = dependencySet.stream()
-                .anyMatch { dependenciesEqual(it, dependency) }
-        if (!isDuplicate) {
-            dependencySet.add(dependency)
+    /**
+     * A map that contains the relations of known Gradle configuration names
+     * to their Maven dependency scope equivalents.
+     */
+    private static Map<String, DependencyScope> CONFIG_TO_SCOPE
```

Comment:
Let's move this declaration to go after `final` fields so that its static initialization starts right after the declaration.